sdk/monitor/azure-monitor-opentelemetry-exporter/src/main/java/com/azure/monitor/opentelemetry/exporter/implementation/utils/ResourceParser.java

```diff
@@ -41,9 +41,13 @@ public void setRoleNameAndInstance(
             return;
         }
 
-        builder.addTag(ContextTagKeys.AI_CLOUD_ROLE.toString(), getRoleName(resource));
+        if (!builder.build().getTags().containsKey(ContextTagKeys.AI_CLOUD_ROLE.toString())) {
+            builder.addTag(ContextTagKeys.AI_CLOUD_ROLE.toString(), getRoleName(resource));
+        }
 
-        builder.addTag(ContextTagKeys.AI_CLOUD_ROLE_INSTANCE.toString(), getRoleInstance(resource));
+        if (!builder.build().getTags().containsKey(ContextTagKeys.AI_CLOUD_ROLE_INSTANCE.toString())) {
+            builder.addTag(ContextTagKeys.AI_CLOUD_ROLE_INSTANCE.toString(), getRoleInstance(resource));
+        }
```

Comment:
this code (`getRoleName`/`getRoleInstance`) isn't needed from Java agent anyways (since they are calculated in ConfigurationBuilder), so I'd recommend inlining this part only into the Java agent instead of calling `setRoleNameAndInstance`

```
if (AksResourceAttributes.isAks(resource)) {
            builder.addTag(ContextTagKeys.AI_CLOUD_ROLE.toString(), AksResourceAttributes.getAksRoleName(resource));
            builder.addTag(ContextTagKeys.AI_CLOUD_ROLE_INSTANCE.toString(), AksResourceAttributes.getAksRoleInstance(resource));
            return;
        }
```

Reply:
nm, this is good